### PR TITLE
POWER CONSUMPTION now depends on the PLAYTIME OF ENGINEERS 

### PR DESCRIPTION
**Pull request by Simplehorror**
# Document the changes in your pull request
To make it easier to balance around having two different kinds of engineers, noob and pro, I have decided to make it easier for noob engineers to power the station and harder for pro engineers to power the station, which means the hours you learn how to make a 1000 TW supermatter are actually useful.

# Wiki Documentation
Engineering gets HARDER NOW

# Changelog

:cl:  
rscadd: POWER CONSUMPTION now depends on the PLAYTIME OF ENGINEERS 
/:cl:

### CODE REVIEW

**Comment by LoliconSlayer** [comment]
Cringe

**Comment by Simplehorror** [comment]
> Cringe

name one thing wrong with this PR or idea

**Comment by ShoziOffical** [comment]
X2

**Comment by ShoziOffical** [comment]
This does not make any sense

**Comment by Simplehorror** [comment]
> X2

what emoticon is this

**Comment by Simplehorror** [comment]
> This does not make any sense

???

**Comment by LoliconSlayer** [comment]
> > Cringe
> 
> name one thing wrong with this PR or idea

The fact that you made it

**Comment by patpol4** [comment]
Crazy

**Comment by Simplehorror** [comment]
> > > Cringe
> > 
> > 
> > name one thing wrong with this PR or idea
> 
> The fact that you made it

ad hominem

**Comment by LoliconSlayer** [comment]
> > > > Cringe
> > > 
> > > 
> > > name one thing wrong with this PR or idea
> > 
> > 
> > The fact that you made it
> 
> ad hominem

Correct, want a cookie?

**Comment by Simplehorror** [comment]
engineer mains on suicide watch after actually having to do more than look at the SM to fully power the station

**Comment by Bibby0110** [comment]
We’re not doing this.

In addition stop putting weird UTF characters in your PRs, and start providing proper descriptions

**Comment by Simplehorror** [comment]
> We’re not doing this.
> 
> In addition stop putting weird UTF characters in your PRs, and start providing proper descriptions

why are we not doing this????

**Comment by Simplehorror** [comment]
https://www.youtube.com/watch?v=I3izqJM44HI

**Comment by The-Sun-In-Splendour** [comment]
have u thought that maybe veteran engineers sometimes dont want to spend half the round making an autismo engine and jut want to hook up N2 and do whatever else instead???? have you thought of that????

**Comment by The-Sun-In-Splendour** [comment]
oh btw stop with your weird ass unicode characters in your pr names they make me feel like im reading a txt file inside an installer to pirate a game with like "PIRATED BY XXX_HACKERGROUP_XXX"

**Comment by Simplehorror** [comment]
> have u thought that maybe veteran engineers sometimes dont want to spend half the round making an autismo engine and jut want to hook up N2 and do whatever else instead???? have you thought of that????

yeah because right now there's no point in doing more than N2

**Comment by Simplehorror** [comment]
> oh btw stop with your weird ass unicode characters in your pr names they make me feel like im reading a txt file inside an installer to pirate a game with like "PIRATED BY XXX_HACKERGROUP_XXX"

it sure caught your attention

**Comment by ShoziOffical** [comment]
> > oh btw stop with your weird ass unicode characters in your pr names they make me feel like im reading a txt file inside an installer to pirate a game with like "PIRATED BY XXX_HACKERGROUP_XXX"
> 
> it sure caught your attention

This is not a title competition, you are here to suggest changes and people here will discuss about it

File: code/modules/jobs/job_types/station_engineer.dm [hunk 1]
```diff
@@ -94,6 +94,27 @@ GLOBAL_LIST_INIT(available_depts_eng, list(ENG_DEPT_MEDICAL, ENG_DEPT_SCIENCE, E
...
+	
+	//Time to balance
+
+	var/x1 = 10
+	var/x2 = most_hours_played
+	var/x3 = 1000
+	var/y1 = 0.5
+	var/y3 = 120
+
+	var/a1 = (((x2 - x1) * (y3 - y1)) / (x3 - x1)) + y1
```
Comment (Notamaniac) [inline code/modules/jobs/job_types/station_engineer.dm:114]: what in the fuck is this magical formula

I'd understand if it was some sort of exponential or logarithmic curve. But it's not.

**Comment by Notamaniac** [comment]
So just doing the rough maths, an engineer with 300 hours would result in a total multiplier of ((300-10)*(120-0.5)/(1000-10)) + 0.5 = 35.51. With the average power consumption of the station being approximately 250 KW, give or take, it would mean you would need to provide approximately 8.775 MW to power the station fully.

For reference, that is approximately 98 sets of sixty solar panels solars operating constantly (90KW per array) ( which they don't, so it would be closer to 196 sets of solars), 35 supermatters using a round start setup and full plasma tanks, (250 KW approx from each super matter), 14.6 TEG's with the roundstart setup (600 KW roughly per engine), 8 tesla's shocking 6 unupgraded tesla coils, or 66 radiation collectors loaded with full plasma tanks around a level 3 singularity (rough numbers, 800 KW per every six rad collectors).

In short, very good well balanced pr. Experienced engineers will love the extra tedium of setting up 35 super matters to keep up with station demand. Or 4 maybe if you use something like a carbon setup and leave a BSRPED with 100 parts in it.

This also doesn't take in account if your engineer is a traitor or even worse a darkspawn or shadowling, which would leave power generation to a less experienced player, who would probably not know how to make a 1000 TW supermatter, or might not even know how to set ot up at all. Much less if the experienced engineer immediately suicides, goes SSD, fucks off to nowhere, or gets rushed by a tator/ling and dies.

**Comment by Sniblet** [comment]
I think I have like 400 hours as engineer and I've never learned anything but the N2 setup
And I absolutely do not want to be forced to
Sorry, I just don't like atmos

This also means grid sabotage leads to faster blackouts the longer the engineers have been playing

I don't like that this encourages people to make new accounts or transfer into engineering after roundstart so that they don't accidentally grief by existing

**Comment by knightshade0** [comment]
I also want to point out- it's hell trying to onboard new engineers as it is. All this does it make it even harder.

**Comment by CMOisLing** [comment]
Imagine the only 1000 hour engineer on your newbie team is tator and fucks off. So you have to deal with making a pure trit engine to run the station, while your still figuring out how to construct a window.

**Comment by Simplehorror** [comment]
> Imagine the only 1000 hour engineer on your newbie team is tator and fucks off. So you have to deal with making a pure trit engine to run the station, while your still figuring out how to construct a window.

OH yeah i forgot to account for traitors i kept thinking i was missing something

**Comment by Simplehorror** [comment]
> I think I have like 400 hours as engineer and I've never learned anything but the N2 setup And I absolutely do not want to be forced to Sorry, I just don't like atmos
> 
> This also means grid sabotage leads to faster blackouts the longer the engineers have been playing
> 
> I don't like that this encourages people to make new accounts or transfer into engineering after roundstart so that they don't accidentally grief by existing

then don't play engi if you cant setup the engine properly

**Comment by Simplehorror** [comment]
> So just doing the rough maths, an engineer with 300 hours would result in a total multiplier of ((300-10)*(120-0.5)/(1000-10)) + 0.5 = 35.51. With the average power consumption of the station being approximately 250 KW, give or take, it would mean you would need to provide approximately 8.775 MW to power the station fully.
> 
> For reference, that is approximately 98 sets of sixty solar panels solars operating constantly (90KW per array) ( which they don't, so it would be closer to 196 sets of solars), 35 supermatters using a round start setup and full plasma tanks, (250 KW approx from each super matter), 14.6 TEG's with the roundstart setup (600 KW roughly per engine), 8 tesla's shocking 6 unupgraded tesla coils, or 66 radiation collectors loaded with full plasma tanks around a level 3 singularity (rough numbers, 800 KW per every six rad collectors).
> 
> In short, very good well balanced pr. Experienced engineers will love the extra tedium of setting up 35 super matters to keep up with station demand. Or 4 maybe if you use something like a carbon setup and leave a BSRPED with 100 parts in it.
> 
> This also doesn't take in account if your engineer is a traitor or even worse a darkspawn or shadowling, which would leave power generation to a less experienced player, who would probably not know how to make a 1000 TW supermatter, or might not even know how to set ot up at all. Much less if the experienced engineer immediately suicides, goes SSD, fucks off to nowhere, or gets rushed by a tator/ling and dies.

you can literally provide 8 MW with a CO2 SM without cheesing it with stock parts, or one TEG

File: code/modules/jobs/job_types/station_engineer.dm (hunk 1 above)
Comment (Simplehorror) [reply code/modules/jobs/job_types/station_engineer.dm:114]: linear interpolation

Comment (tattax) [reply code/modules/jobs/job_types/station_engineer.dm:114]: could you put a comment or a better name on what the vars are doing and why. Defines would help.

**Comment by CMOisLing** [comment]
> OH yeah i forgot to account for traitors i kept thinking i was missing something

I think you've forgotten about more than just that. What about roundstart? station is going to blackout much quicker and mega autism engines take longer to setup. What about your other PR with stock parts with some machines taking 31MW to power? How are shocks going to be affected by this? What about charlie station and other ghost roles power stations? What about the CE? What if atmos is incompetent? What if an early conversion antag take over engineering? Sling balance?

**Comment by Simplehorror** [comment]
> > OH yeah i forgot to account for traitors i kept thinking i was missing something
> 
> I think you've forgotten about more than just that. What about roundstart? station is going to blackout much quicker and mega autism engines take longer to setup. What about your other PR with stock parts with some machines taking 31MW to power? How are shocks going to be affected by this? What about charlie station and other ghost roles power stations? What about the CE? What if atmos is incompetent? What if an early conversion antag take over engineering? Sling balance?

shocks unaffected

**Comment by CMOisLing** [comment]
> shocks unaffected
You need more power into the grid to keep everything charged this directly affects shocks

**Comment by Simplehorror** [comment]
> > shocks unaffected
> > You need more power into the grid to keep everything charged this directly affects shocks

anti-greytide technology

**Comment by Necrosium** [comment]
This looks like a joke PR from every single point of view. I literally fail to see how this is going to help engineers in any way.

**Comment by Simplehorror** [comment]
> This looks like a joke PR from every single point of view. I literally fail to see how this is going to help engineers in any way.

it's not meant to help them, it's called dynamic difficulty and every game since 1999 has had it

**Comment by Sniblet** [comment]
> it's called dynamic difficulty and every game since 1999 has had it

Does this include competitive multiplayer games and role playing games in which player interactions, as much as or more than mechanical challenges, are the focus
I remember Halo Wars AI having dynamic difficulty. You could turn it off, by the way

If someone wants to generate 40 quintillion exawatts with half an SM shard they are welcome to try as is, their reward is having done it and taken the screenshots. That’s some positive reinforcement right there. That makes the game fun. It makes players feel good
Forcing them to do so every round or else ruin the station for everyone else on account of having wrongly taken up the massive responsibility of station engineer is negative reinforcement, it punishes players for
For, get this
_Playing a role too much_
Or even _being in the same round as someone else who did_
It makes the job (which is part of a game (games are supposed to be fun)) more unpleasant
It makes most decent people
Not want to play engineer
It makes people who want to play optimally
Not want to play engineer
In a word, you are making engineer… worse

Am I making sense to you yet
If not, I can start getting into the several extraneous gameplay effects of this change that you don’t appear to have considered instead

**Comment by Simplehorror** [comment]
> > it's called dynamic difficulty and every game since 1999 has had it
> 
> Does this include competitive multiplayer games and role playing games in which player interactions, as much as or more than mechanical challenges, are the focus I remember Halo Wars AI having dynamic difficulty. You could turn it off, by the way
> 
> If someone wants to generate 40 quintillion exawatts with half an SM shard they are welcome to try as is, their reward is having done it and taken the screenshots. That’s some positive reinforcement right there. That makes the game fun. It makes players feel good Forcing them to do so every round or else ruin the station for everyone else on account of having wrongly taken up the massive responsibility of station engineer is negative reinforcement, it punishes players for For, get this _Playing a role too much_ Or even _being in the same round as someone else who did_ It makes the job (which is part of a game (games are supposed to be fun)) more unpleasant It makes most decent people Not want to play engineer It makes people who want to play optimally Not want to play engineer In a word, you are making engineer… worse
> 
> Am I making sense to you yet If not, I can start getting into the several extraneous gameplay effects of this change that you don’t appear to have considered instead

you're not making sense because your point is an appeal to emotion. when i play engineer i don't bother doing my best because it isn't required, but if the station needed it i would gladly do co2 sm every round because that's the new baseline for me

**Comment by Notamaniac** [comment]
> you're not making sense because your point is an appeal to emotion. when i play engineer i don't bother doing my best because it isn't required, but if the station needed it i would gladly do co2 sm every round because that's the new baseline for me

Let me highlight something in that statement.
>**i would gladly**

Yes. YOU would gladly. Have you heard from literally anyone else about them liking this change? Exactly how many no's are you getting to how many yes's? You are aware that the things **you** like aren't the things that everyone else would like?

Why should engineers with 300 hours be forced to break into atmos for CO2 SM/ rig up trit burns for a TEG? Because **you** would like it.

Well fuck if I can justify any argument based on the things I like, then everyone should spawn in with infinite ammo pulse rifles and infinite grenades. Medical should have insta heal sleepers, and ignore the damage cap on revives. Science should have infinite research points, and security should just not exist. Except I wouldn't add these changes because I know that would ruin the game for other people.

It's an idea that a lot of people don't like. Just let it go.

**Comment by TimeURSS** [comment]
> > This looks like a joke PR from every single point of view. I literally fail to see how this is going to help engineers in any way.
> 
> it's not meant to help them, it's called dynamic difficulty and every game since 1999 has had it

So you simply want the server to not have experienced engineers at all. 

That PR, if it gets merged, not only will make people experienced in engineering to think twice before readying up as engineers, but will just leave the inexperienced ones that do take the role to set up potentially faulty engines that could lead to the same issue: no power.

I would not like to, or see someone take a random role roundstart and then powergame engineering just to avoid this thing.

**Comment by Sniblet** [comment]
> you're not making sense because your point is an appeal to emotion. when i play engineer i don't bother doing my best because it isn't required, but if the station needed it i would gladly do co2 sm every round because that's the new baseline for me

If you sincerely believe that an appeal to the emotion of “having fun” is meaningless in a video game’s context, I don’t know how to talk to you.
> name one thing wrong with this PR or idea

1. Network admin no longer functions in rounds with at least a 2x consumption multiplier (out of a maximum 100x, if I’m reading your math right). An upgraded AI’s power consumption already takes a heavy toll on most power grids, and now it takes more than any single engine can supply. Upgrading the AI on any significant multiplier is now grief.
2. Charlie station, the Syndicate Bioweapons Lab, the mining outpost, and a few space ruins no longer function on any significant modifier. Charlie station doesn't get to find out that they can't play until they've already wired up their solars and had nothing happen.
3. Telecomms will have to be hotwired in the event of a >19x multiplier.
4. Powersinks may explode much faster.
5. Grid cutouts can lead to powersink-like effects with how fast rooms black out, especially chem, atmos and the hotwired telecoms.
6. Solars no longer matter.
7. Singularities are now a waste of a containment pen, as despite the rad collector buff, you just can’t run more than one in a given pen.
8. Cargo sometimes has to pour extreme amounts of money and energy into supplying more generators.
9. There is nothing interesting, unusual, or impressive about making a strong engine anymore (appeal to emotion, sorry)
10. It’s much more difficult to judge when a room is taking too much power, as there is no consistent baseline
11. BSAs no longer function if the modifier substantially exceeds 1, as no power cell big enough to supply their APC is accessible to players
12. Dark antags and AI killers have a field day every day the good engineers/engineers with a lot of AFK time are on, which is the opposite of how it should work
13. War ops and clock cult, gamemodes where any engine at all usually becomes unnecessary, may now require a very strong engine to be made very fast, demanding all of engineering's attention for most of the round (which is very cool for the coggers)
14. Not knowing how to run a CO2 SM as an engineer becomes a rules issue
15. Admins may have to intervene with void cores and infinite power cells to prevent severe round disruption caused by players playing too much
16. It can become much harder, actually, to generate enough excess power to shock people very badly
17. The emitter SMES may need to be upgraded at roundstart
18. Stationwide blackouts at the start of any round where someone has too much playtime are to be expected, making their playtime everyone’s problem
19. New engineers are placed under a timer to learn as much about power generation as possible before they hit that liability threshold
20. Existing engineers with a lot of playtime but not much super-engine experience (me) now must never play engineer again
21. All engineers are incentivized to not play engineer or to regularly switch to new accounts for that sweet 0.5x multiplier so they can use the excessive excess power and greatly increased grace period on responding to grid cuts, to do actually fun things with their excess power and do anything besides tending the power grid
22. Engineers who join as engineer too much are correspondingly heckled for making the power situation monumentally worse
23. Joining as engineer is now objectively harmful to the station in a way that most antagonists cannot match

So what was the upside again?

**Comment by LoliconSlayer** [comment]
Gonna quick note that pointing out logical fallacies doesn't make you smart. Additionally, just because an argument contains a fallacy does not mean the argument is automatically incorrect, pointing out fallacies as if doing so automatically invalidates an argument is actually an example of the fallacy fallacy.

**Comment by CMOisLing** [comment]
The is the best troll PR ever. 10/10 completely fooled me.

**Comment by Simplehorror** [comment]
Closed in favour of #16424, the yogs playerbase clearly isn't ready for dynamic difficulty.

**Comment by Simplehorror** [comment]
#16424 was closed, so reopened

**Comment by N3D6** [comment]
This could quite possibly be the worst pr I’ve ever seen

**Comment by Simplehorror** [comment]
image

**Comment by Bibby0110** [comment]
I think you might have accidentally clicked the reopen button, let me help you out :)
